src/components/UsersCountModal/index.tsx

```diff
@@ -0,0 +1,111 @@
+import classNames from 'classnames';
+import React, { ReactElement, useEffect, useState } from 'react';
+
+import {
+  fetchLikersByPost,
+  fetchUserFollowers,
+  fetchUserFollowings,
+  useMeta,
+} from '../../ducks/posts';
+
+import Modal, { ModalContent, ModalHeader } from '../Modal';
+import { UserRow } from '../DiscoverUserPanel';
+import { useUser } from '../../ducks/users';
+
+function UsersList(props: { onClose: () => void; users: string[]; title: string }): ReactElement {
+  const { onClose, title, users } = props;
+  return (
+    <Modal className={classNames('w-148')} onClose={onClose}>
+      <ModalHeader>{title}</ModalHeader>
+      <ModalContent>
+        {users.map(ens => (
+          <UserRow key={ens} name={ens} />
+        ))}
+      </ModalContent>
+    </Modal>
+  );
+}
+
+export enum Item {
+  Like = 'Like',
+  Follower = 'Follower',
+  Following = 'Following',
+}
+
+const useItem = (item: Item, id: string) => {
+  if (item === Item.Like) return { fetch: fetchLikersByPost, count: useMeta(id).likeCount };
+  if (item === Item.Follower)
+    return { fetch: fetchUserFollowers, count: useUser(id)?.meta.followerCount };
+
+  return { fetch: fetchUserFollowings, count: useUser(id)?.meta.followingCount };
+};
+
+function MaybePlural(props: { users: string[]; text: Item }): ReactElement {
+  const { users, text } = props;
+  const count = users.length;
+  const many = count > 1 && text !== Item.Following;
+
+  return (
+    <div className="flex flex-row flex-nowrap item-center">
+      <div className="font-semibold">{count}</div>
+      <div className="ml-2 text-gray-500">{`${text}${many ? 's' : ''}`}</div>
+    </div>
+  );
+}
+
+export default function UsersCountModal(props: {
+  className: string;
+  item: Item;
+  id: string;
+}): ReactElement {
+  const { className, item, id } = props;
+  const [showingList, setShowList] = useState(false);
+
+  const [users, setUsers] = useState<string[] | null>(null);
+
+  // hacky?
+  const { fetch, count } = useItem(item, id);
+
+  const many = count && count > 1 && item !== Item.Following;
+  const text = `${item}${many ? 's' : ''}`;
+  const title = item === Item.Like ? 'Liked By' : text;
+
+  useEffect(() => {
+    setShowList(false);
+    fetch(id).then(users => setUsers(users));
```

Comment:
This returned a wrong list (null) when reacting to a click on the like button.
I haven't managed to fix it. See last 10s of https://www.loom.com/share/f3707bd1813b4a2b988ab5ac3b377d5e

- click on like
- `count` gets updated
- it's a dep of the useEffect hook so we refetch
- still the list of users returned is `null`
- so the modal isn't displayed (guard on `users`)

Reply:
@0xtsukino, I could fix this by moving the fetching into the `onClick` callback like you suggested. See 55b8a50.